rackcity/views/it_model_views.py

```diff
@@ -18,22 +18,33 @@ def model_list(request):
         return JsonResponse(serializer.data, safe=False)
 
 
-def model_add(request):  # TODO: make it only accept POSTs
+@api_view(['POST'])
+@permission_classes([IsAdminUser])
+def model_add(request):
     """
     Add a new model
     """
-    if request.method == 'POST':
-        data = JSONParser().parse(request)
-        serializer = ITModelSerializer(data=data)
-        if serializer.is_valid():  # TODO: figure out what .valid() does.  Different fields are required for add, update, etc
-            print("valid")
+    data = JSONParser().parse(request)
+    failure_message = ""
+    if 'id' in data:
+        failure_message = failure_message + "Don't include id when adding a model. "
```

Comment:
do we want the request to fail if `id` is passed as a param? your current implementation allows this to continue on to the serialize attempt

Reply:
So it would try to validate the object which I think is good because you'll get a more complete error message if there are other problems, but then it won't try to save the model because the error message is nonempty